docs/reference/contributing/guidelines/workflow.md

```diff
@@ -61,90 +61,142 @@ If commits do not follow the above guidelines, we may request that you modify th
-- Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
+#### Patch update
 
-Release: feature
+This can contain any of the following changes:
 
-### Documentation update
+##### Bug Fix
 
-Documentation updates include changes to markdown files or Doxygen documentation (comment-only changes).
+A bug fix is a change that fixes a specific defect in the codebase with backward compatibility. These are the highest priority because of the positive effect the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Every bug fix should contain a test to verify results before and after the pull request. Bug fixes are candidates for patch releases. Large, nontrivial bug fixes approaching the size of refactors run the risk of being considered refactors themselves.
+
+##### Refactor
+
+A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this are moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes and test organization changes. There is always the risk that someone depended on the location or name before a refactor; therefore, these are lower in priority than bug fixes and might require detailed justification for the change. Refactors are candidates for feature releases.
+
+##### Target update
+
+Updating target implementation (adding a new target or updating already supported target) is a change for a patch release.
+
+A test report for the new target must be part of the pull request. The new target must pass all Mbed OS functional and system validation tests (using `mbed test` command) for the current Mbed OS major release, including all Mbed OS supported toolchains.
+
+##### Documentation update
 
-Release: patch
+Documentation updates include changes to markdown files or Doxygen documentation (comment-only changes).
 
-### Test update
+##### Test update
 
 Test updates include updates to a new test unit or test case.
 
-Release: patch
+#### Feature update
+
+This can be any change in the functionality, including adding a new feature, a new method or a function. Software language does not matter.
+
+A feature contribution contains a new API, capability or behavior. It does not break backward compatibility with existing APIs, capabilities or behaviors. New feature contributions are very welcome in Mbed OS. However, because they add capability to the codebase, a new feature may introduce bugs and a support burden. New features should also come with documentation, support for most targets and comprehensive test coverage. Feature PRs are treated cautiously, and new features require a new minor version for the codebase. Features are candidates for feature releases.
```

Comment:
should this contain also this: "The Mbed OS technical lead must approve feature change pull requests.", this is currently what we do

Reply:
Yes we can add that.